### ISSUE TITLE
[false positive] "Unsupported converter" raised for methods

### ISSUE THREAD

**Issue by alexchandel** [python/mypy#10733]
**Bug Report**

`mypy` reports an error when a method is used as an `attr` converter.

**To Reproduce**

```python
import attr
@attr.s(auto_attribs=True, slots=True)
class UpperString():
    """Uppercase string."""
    value: str = attr.ib(converter=str.upper)
```

**Expected Behavior**

`str.upper` is a named method.  It takes a `str` (self, usually) and returns a `str.

**Actual Behavior**

"Unsupported converter, only named functions and types are currently supported"

**Your Environment**

- Mypy version used: 0.910 (latest)
- Mypy command-line flags: N/A
- Mypy configuration options from `mypy.ini` (and other config files): N/A
- Python version used: latest
- Operating system and version: all
